File: src/main/java/com/sprint/BookPartnerApplication/services/SalesService.java
package com.sprint.BookPartnerApplication.services;

import com.sprint.BookPartnerApplication.dto.request.SalesRequestDTO;
import com.sprint.BookPartnerApplication.entity.Sales;
import java.time.LocalDateTime;
import java.util.List;

public interface SalesService {

    Sales createSale(SalesRequestDTO dto);          // POST
    List<Sales> getAllSales();                       // GET all
    List<Sales> getSalesByStore(String storeId);    // GET by store
    List<Sales> getSalesByTitle(String titleId);    // GET by title
    List<Sales> getSalesByDateRange(LocalDateTime from, LocalDateTime to); // GET by date
}

File: BookPartner-UI/src/app/service/sales.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';

@Injectable({
  providedIn: 'root'
})
export class SalesService {
  private baseUrl = 'http://localhost:8082/api/v1/sales';

  constructor(private http: HttpClient) {}

  createSale(dto: any): Observable<any> {
    return this.http.post(`${this.baseUrl}`, dto);
  }

  getAllSales(): Observable<any[]> {
    return this.http.get<any[]>(`${this.baseUrl}`);
  }

  getSalesByStore(storeId: string): Observable<any[]> {
    return this.http.get<any[]>(`${this.baseUrl}/store/${storeId}`);
  }

  getSalesByTitle(titleId: string): Observable<any[]> {
    return this.http.get<any[]>(`${this.baseUrl}/title/${titleId}`);
  }

  getSalesByDateRange(from: string, to: string): Observable<any[]> {
    const params = new HttpParams().set('from', from).set('to', to);
    return this.http.get<any[]>(`${this.baseUrl}/date-range`, { params });
  }
}